File: libs/eave-stdlib-py/src/eave/stdlib/core_api/operations/github_documents.py
from typing import Optional, Unpack
import uuid

from eave.stdlib.core_api.models.github_repos import GithubRepo, GithubRepoRefInput
from ... import requests
from eave.stdlib.core_api.models.github_documents import (
    GithubDocument,
    GithubDocumentsDeleteByIdsInput,
    GithubDocumentsDeleteByTypeInput,
    GithubDocumentsQueryInput,
    GithubDocumentCreateInput,
    GithubDocumentUpdateInput,
)
from . import BaseRequestBody, BaseResponseBody, CoreApiEndpoint, CoreApiEndpointConfiguration


class GetGithubDocumentsRequest(CoreApiEndpoint):
    config = CoreApiEndpointConfiguration(
        path="/github-documents/query",
        auth_required=False,
    )

    class RequestBody(BaseRequestBody):
        query_params: GithubDocumentsQueryInput

    class ResponseBody(BaseResponseBody):
        documents: list[GithubDocument]

    @classmethod
    async def perform(
        cls,
        input: RequestBody,
        team_id: uuid.UUID | str,
        account_id: Optional[uuid.UUID],
        access_token: Optional[str],
        **kwargs: Unpack[requests.CommonRequestArgs],
    ) -> ResponseBody:
        response = await requests.make_request(
            config=cls.config,
            input=input,
            team_id=team_id,
            account_id=account_id,
            access_token=access_token,
            **kwargs,
        )

        body = await cls.make_response(response, cls.ResponseBody)
        return body


class CreateGithubDocumentRequest(CoreApiEndpoint):
    config = CoreApiEndpointConfiguration(
        path="/github-documents/create",
        auth_required=False,
    )

    class RequestBody(BaseRequestBody):
        repo: GithubRepoRefInput
        document: GithubDocumentCreateInput

    class ResponseBody(BaseResponseBody):
        repo: GithubRepo
        document: GithubDocument

    @classmethod
    async def perform(
        cls,
        input: RequestBody,
        team_id: uuid.UUID,
        account_id: Optional[uuid.UUID],
        access_token: Optional[str],
        **kwargs: Unpack[requests.CommonRequestArgs],
    ) -> ResponseBody:
        response = await requests.make_request(
            config=cls.config,
            input=input,
            team_id=team_id,
            account_id=account_id,
            access_token=access_token,
            **kwargs,
        )

        body = await cls.make_response(response, cls.ResponseBody)
        return body


class UpdateGithubDocumentRequest(CoreApiEndpoint):
    config = CoreApiEndpointConfiguration(
        path="/github-documents/update",
        auth_required=False,
    )

    class RequestBody(BaseRequestBody):
        document: GithubDocumentUpdateInput

    class ResponseBody(BaseResponseBody):
        document: GithubDocument

    @classmethod
    async def perform(
        cls,
        input: RequestBody,
        team_id: uuid.UUID,
        account_id: Optional[uuid.UUID],
        access_token: Optional[str],
        **kwargs: Unpack[requests.CommonRequestArgs],
    ) -> ResponseBody:
        response = await requests.make_request(
            config=cls.config,
            input=input,
            team_id=team_id,
            account_id=account_id,
            access_token=access_token,
            **kwargs,
        )

        body = await cls.make_response(response, cls.ResponseBody)
        return body


class DeleteGithubDocumentsByIdsRequest(CoreApiEndpoint):
    config = CoreApiEndpointConfiguration(
        path="/github-documents/delete/id",
        auth_required=False,
    )

    class RequestBody(BaseRequestBody):
        documents: list[GithubDocumentsDeleteByIdsInput]

    class ResponseBody(BaseResponseBody):
        pass

    @classmethod
    async def perform(
        cls,
        input: RequestBody,
        team_id: uuid.UUID,
        account_id: Optional[uuid.UUID],
        access_token: Optional[str],
        **kwargs: Unpack[requests.CommonRequestArgs],
    ) -> ResponseBody:
        response = await requests.make_request(
            config=cls.config,
            input=input,
            account_id=account_id,
            access_token=access_token,
            team_id=team_id,
            **kwargs,
        )

        body = await cls.make_response(response, cls.ResponseBody)
        return body


class DeleteGithubDocumentsByTypeRequest(CoreApiEndpoint):
    config = CoreApiEndpointConfiguration(
        path="/github-documents/delete/type",
        auth_required=False,
    )

    class RequestBody(BaseRequestBody):
        documents: GithubDocumentsDeleteByTypeInput

    class ResponseBody(BaseResponseBody):
        pass

    @classmethod
    async def perform(
        cls,
        input: RequestBody,
        team_id: uuid.UUID,
        account_id: Optional[uuid.UUID],
        access_token: Optional[str],
        **kwargs: Unpack[requests.CommonRequestArgs],
    ) -> ResponseBody:
        response = await requests.make_request(
            config=cls.config,
            input=input,
            team_id=team_id,
            account_id=account_id,
            access_token=access_token,
            **kwargs,
        )

        body = await cls.make_response(response, cls.ResponseBody)
        return body

File: libs/eave-stdlib-ts/src/core-api/operations/github-documents.ts
import { makeRequest, RequestArgsTeamId } from "../../requests.js";
import {
  GithubDocument,
  GithubDocumentCreateInput,
  GithubDocumentsDeleteByIdsInput,
  GithubDocumentsDeleteByTypeInput,
  GithubDocumentsQueryInput,
  GithubDocumentUpdateInput,
} from "../models/github-documents.js";
import { GithubRepo, GithubRepoRefInput } from "../models/github-repos.js";
import { CoreApiEndpointConfiguration } from "./shared.js";

export type GetGithubDocumentsRequestBody = {
  query_params: GithubDocumentsQueryInput;
};

export type GetGithubDocumentsResponseBody = {
  documents: Array<GithubDocument>;
};

export class GetGithubDocumentsOperation {
  static config = new CoreApiEndpointConfiguration({
    path: "/github-documents/query",
  });
  static async perform(
    args: RequestArgsTeamId & { input: GetGithubDocumentsRequestBody },
  ): Promise<GetGithubDocumentsResponseBody> {
    const resp = await makeRequest({
      config: this.config,
      ...args,
    });
    const responseData = <GetGithubDocumentsResponseBody>await resp.json();
    return responseData;
  }
}
export type CreateGithubDocumentRequestBody = {
  repo: GithubRepoRefInput;
  document: GithubDocumentCreateInput;
};

export type CreateGithubDocumentResponseBody = {
  repo: GithubRepo;
  document: GithubDocument;
};

export class CreateGithubDocumentOperation {
  static config = new CoreApiEndpointConfiguration({
    path: "/github-documents/create",
  });
  static async perform(
    args: RequestArgsTeamId & { input: CreateGithubDocumentRequestBody },
  ): Promise<CreateGithubDocumentResponseBody> {
    const resp = await makeRequest({
      config: this.config,
      ...args,
    });
    const responseData = <CreateGithubDocumentResponseBody>await resp.json();
    return responseData;
  }
}
export type UpdateGithubDocumentRequestBody = {
  document: GithubDocumentUpdateInput;
};

export type UpdateGithubDocumentResponseBody = {
  document: GithubDocument;
};

export class UpdateGithubDocumentOperation {
  static config = new CoreApiEndpointConfiguration({
    path: "/github-documents/update",
  });
  static async perform(
    args: RequestArgsTeamId & { input: UpdateGithubDocumentRequestBody },
  ): Promise<UpdateGithubDocumentResponseBody> {
    const resp = await makeRequest({
      config: this.config,
      ...args,
    });
    const responseData = <UpdateGithubDocumentResponseBody>await resp.json();
    return responseData;
  }
}
export type DeleteGithubDocumentsByIdsRequestBody = {
  documents: Array<GithubDocumentsDeleteByIdsInput>;
};

export class DeleteGithubDocumentsByIdsOperation {
  static config = new CoreApiEndpointConfiguration({
    path: "/github-documents/delete/id",
  });
  static async perform(
    args: RequestArgsTeamId & {
      input: DeleteGithubDocumentsByIdsRequestBody;
    },
  ): Promise<void> {
    await makeRequest({
      config: this.config,
      ...args,
    });
  }
}
export type DeleteGithubDocumentsByTypeRequestBody = {
  documents: Array<GithubDocumentsDeleteByTypeInput>;
};

export class DeleteGithubDocumentsByTypeOperation {
  static config = new CoreApiEndpointConfiguration({
    path: "/github-documents/delete/type",
  });
  static async perform(
    args: RequestArgsTeamId & {
      input: DeleteGithubDocumentsByTypeRequestBody;
    },
  ): Promise<void> {
    await makeRequest({
      config: this.config,
      ...args,
    });
  }
}
